src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,137 @@
+import classNames from 'classnames';
+
+import React, { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+import { onUpdate } from '../../api/todos';
+
+type Props = {
+  todo: Todo,
+  withLoader?: boolean,
+  onRemoveTodo: (id: number) => void,
+  onCompletedChange: (value: Todo) => void,
+  loadTodos: () => void,
+  setErrorMessage:(value: string) => void,
+};
+
+export const TodoItem: React.FC<Props> = React.memo(
+  ({
+    todo,
+    withLoader,
+    onRemoveTodo,
+    onCompletedChange,
+    loadTodos,
+    setErrorMessage,
+  }) => {
+    const { id, completed, title } = todo;
+
+    const [hasLoader, setHasLoader] = useState(withLoader);
+    const [isDoubleClick, setIsDoubleClick] = useState(false);
+
+    const [editedTitle, setEditedTitle] = useState(title);
+
+    const handleDelete = () => {
+      onRemoveTodo(id);
+      setHasLoader(true);
+    };
+
+    const handleClick = async () => {
+      setHasLoader(true);
+
+      await onCompletedChange(todo);
+      setHasLoader(false);
+    };
+
+    const onEditTodo = (event: React.ChangeEvent<HTMLInputElement>) => {
+      const { value } = event.target;
+
+      setEditedTitle(value);
+    };
+
+    const onSaveChanges = async (currentTodo: Todo) => {
+      setHasLoader(true);
+      setIsDoubleClick(false);
+
+      if (!editedTitle.length) {
+        handleDelete();
+      }
+
+      try {
+        await onUpdate(currentTodo.id, {
+          id: currentTodo.id,
+          userId: currentTodo.userId,
+          title: editedTitle,
+          completed,
+        });
+      } catch (error) {
+        setErrorMessage('Unable to update a todo');
+      }
+
+      setHasLoader(false);
+      loadTodos();
+    };
+
+    const onKeyPress = async (
+      currentTodo: Todo,
+      event: React.KeyboardEvent,
+    ) => {
+      if (event.key === 'Enter') {
```

Comment:
you should handle `Esc`

![image](https://user-images.githubusercontent.com/86923089/224533151-74a26227-f4d6-41f8-a7e0-aa72d70172e4.png)